main.go

```diff
@@ -524,6 +554,12 @@ func uploadGame(c *gin.Context) {
 		return
 	}
 
+	if !checkPermissionExpr(training_run.PermissionExpr, *user, training_id, c.PostForm("engineVersion"), c.PostForm("version")) {
```

Comment:
Move this up earlier?  Looks like it just needs to be after getTrainingRun rather than after we resolve the network.